### ISSUE TITLE
Non Square Patterns gets sqeezed.

### ISSUE THREAD

**Issue by methelas2** [geftimov/android-patternview#9]
When I try to make widget 3 columns wide and 4 rows high, the rows gets a little bit squeezed. 
Any modification in width/height of the view won't help.

**Comment by geftimov** [comment]
I will look into this issue, thanks for reporting.

**Comment by achinverma** [comment]
yes , i am also facing the same issue when i enter 
patternview:gridColumns="3"
patternview:gridRows="4"

![device-2016-01-29-111741](https://cloud.githubusercontent.com/assets/2811593/12668245/8c71c690-c67a-11e5-8f61-6bd6525131ba.png)

**Comment by methelas2** [comment]
Same thing actually happens even for square patterns but on a lesser scale.

I made my way around it by extending PatternView and overriding onMeasure/onSizeChanged methods as seen below:

`
public class SquarePatternView extends PatternView {

```
    public SquarePatternView(Context context) {
        super(context);
    }

    public SquarePatternView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public SquarePatternView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        int widthSize = MeasureSpec.getSize(widthMeasureSpec);
        int heightSize = MeasureSpec.getSize(heightMeasureSpec);

        int size = widthSize < heightSize ? widthSize : heightSize;

        int finalMeasureSpec = MeasureSpec.makeMeasureSpec(size, MeasureSpec.EXACTLY);
        super.onMeasure(finalMeasureSpec, finalMeasureSpec);
    }

    @Override
    protected void onSizeChanged(final int w, final int h, final int oldw, final int oldh) {

        int size = w < h ? w : h;
        super.onSizeChanged(size, size, oldw, oldh);
    }
}    int widthSize = MeasureSpec.getSize(widthMeasureSpec);
    int heightSize = MeasureSpec.getSize(heightMeasureSpec);

    int size = widthSize < heightSize ? widthSize : heightSize;

    int finalMeasureSpec = MeasureSpec.makeMeasureSpec(size, MeasureSpec.EXACTLY);
    super.onMeasure(finalMeasureSpec, finalMeasureSpec);
}

@Override
protected void onSizeChanged(final int w, final int h, final int oldw, final int oldh) {

    int size = w < h ? w : h;
    super.onSizeChanged(size, size, oldw, oldh);
}
```

}
`

I guess(didn't tried it after all) it may be used to scale the view in any desired proportions by changing values inside those two methods.

**Comment by geftimov** [comment]
Yes you can resize the view but you will loose the option when someone want the exact bounds to be set.
For example width and height : 200dp
There is a bug measuring the image and I can fix it, just do not have the time.

**Comment by geftimov** [comment]
Fix at version [1.0.6](https://github.com/geftimov/android-patternview/releases/tag/1.0.6).
